treeherder/webapp/api/push.py

```diff
@@ -71,6 +70,20 @@ def list(self, request, project):
 
             pushes = pushes.filter(repository=repository)
 
+        search_param = filter_params.get("search")
+        if search_param:
+            filtered_commits = (
+                Commit.objects.annotate(
```

Comment:
This can return three or more push IDs for each commit which is sufficiently old:
1. When the commit landed on autoland.
2. When the commit got added to mozilla-central (merged): Treeherder only stores the last 20 commits of that push.
3. When a push which contains the commit get pushed for the first time to Try.
Are restrictions by repository needed?

Reply:
ah yes, in our app we'll use the project always, so do you think it would be better to add it in the subrequest as well?

Reply:
Yes, if you pass the project as argument, it will prevent the need to handle duplicate revisions from different projects.

Reply:
I'm not sure I understand how to handle this 😅

Reply:
The repository ("project") parameter can be retrieved similar to [this](https://github.com/mozilla/treeherder/blob/e6e08646104d8605457c056853fd0ab95c661e31/treeherder/webapp/api/push.py#L66) which is for requests like  `https://treeherder.mozilla.org/api/project/autoland/push/?full=true&count=10`.